src/test/java/org/sonatype/nexus/ci/iq/IqPolicyEvaluatorIntegrationTest.groovy

```diff
@@ -150,29 +152,59 @@ class IqPolicyEvaluatorIntegrationTest
       jenkins.assertBuildStatusSuccess(build)
   }
 
-  def 'Pipeline build should return null and set status to unstable when build fails with network error'() {
+  @Unroll
+  def 'Pipeline build should return null and set status to unstable when build fails with network error with failBuildOnNetworkError #description'() {
     given: 'a jenkins project'
       WorkflowJob project = jenkins.createProject(WorkflowJob)
       configureJenkins()
 
     when: 'the nexus policy evaluator is executed'
       project.definition = new CpsFlowDefinition('node {\n' +
           'writeFile file: \'dummy.txt\', text: \'dummy\'\n' +
-          'def result = nexusPolicyEvaluation failBuildOnNetworkError: false, iqApplication: \'app\', ' +
+          "def result = nexusPolicyEvaluation ${failBuildOnNetworkErrorScript} iqApplication: \'app\', " +
           'iqStage: \'stage\'\n' +
           'echo \'result-after-failure:\' + result' +
           '}\n')
       def build = project.scheduleBuild2(0).get()
 
     then: 'an exception is thrown when getting proprietary config from IQ server'
-      1 * iqClient.getProprietaryConfigForApplicationEvaluation('app') >> { throw new IOException("BANG!") }
+      1 * iqClient.getProprietaryConfigForApplicationEvaluation('app') >> {
+        throw new IqClientException("ERROR", new IOException("BANG!"))
+      }
 
     then: 'the build status is unstable and the result is null'
       jenkins.assertBuildStatus(Result.UNSTABLE, build)
       build.getLog(100).contains('result-after-failure:null')
+      build.getLog(100).contains('Unable to communicate with IQ Server: ERROR')
+
+    where:
+      description     | failBuildOnNetworkErrorScript
+      'false'         | 'failBuildOnNetworkError: false,'
+      'not supplied'  | ''
+  }
+
+  def 'Pipeline build should set status to fail when build fails with network error with failBuildOnNetworkError true'() {
+    given: 'a jenkins project'
+      WorkflowJob project = jenkins.createProject(WorkflowJob)
+      configureJenkins()
+
+    when: 'the nexus policy evaluator is executed'
+      project.definition = new CpsFlowDefinition('node {\n' +
+          'nexusPolicyEvaluation failBuildOnNetworkError: true, iqApplication: \'app\', iqStage: \'stage\'\n' +
+          '}\n')
+      def build = project.scheduleBuild2(0).get()
+
+    then: 'an exception is thrown when getting proprietary config from IQ server'
+      1 * iqClient.getProprietaryConfigForApplicationEvaluation('app') >> {
+        throw new IqClientException("ERROR", new IOException("BANG!"))
+      }
+
+    then: 'the build status is failure and the error is logged'
+      jenkins.assertBuildStatus(Result.FAILURE, build)
+      build.getLog(100).contains('com.sonatype.nexus.api.exception.IqClientException: ERROR')
```

Comment:
ditto that getLog() will never return null?

Reply:
Yes by contract: 
```public @Nonnull List<String> getLog(int maxLines) throws IOException {```